nucypher/characters/lawful.py

```diff
@@ -1470,11 +1464,26 @@ def encrypt_for_pre(
         )
         return message_kit
 
-    def encrypt_for_dkg(self, plaintext: bytes, conditions: Lingo) -> Ciphertext:
+    def encrypt_for_dkg(
+        self, plaintext: bytes, conditions: Lingo
+    ) -> ThresholdMessageKit:
         validate_condition_lingo(conditions)
-        conditions_bytes = json.dumps(conditions).encode()
-        ciphertext = encrypt(plaintext, conditions_bytes, self.policy_pubkey)
-        return ciphertext
+        conditions_json = json.dumps(conditions)
+        access_conditions = Conditions(conditions_json)
+
+        ciphertext, auth_data = encrypt_for_dkg(
+            plaintext, self.policy_pubkey, access_conditions
+        )
+
+        # authentication for AllowLogic
+        # TODO Replace with `Signer` to be passed as parameter
+        header_hash = keccak_digest(bytes(ciphertext.header))
+        authorization = self.signing_power.keypair.sign(header_hash).to_be_bytes()
```

Comment:
👍🏻 this is currently using an umbral keypair for signing?  In any case this is permitted in the PR due to thoe above stubbed method `is_encryption_authorized` (https://github.com/nucypher/nucypher/issues/3206).  We will address this in a follow-up PR to replace it with a transacting power instead.